pokemon-spring-mvc/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/mvc/controllers/TrainerController.java

```diff
@@ -0,0 +1,211 @@
+     */
+    @RequestMapping(value = "/detail/{id}", method = RequestMethod.GET)
+    public String detail(@PathVariable long id,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriComponentsBuilder) {
+        log.debug("mvc detail({})", id);
+
+        if (trainerFacade.getTrainerWithId(id) == null) {
+            redirectAttributes.addFlashAttribute("alert_danger", String.format(messages.getString("trainer.does.not.exists"), id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+
+        model.addAttribute("trainer", trainerFacade.getTrainerWithId(id));
+        try {
+            model.addAttribute("pokemons", getPokemons(pokemonFacade.getPokemonOfTrainer(id)));
+        } catch (NoSuchEntityException ex) {
+            redirectAttributes.addFlashAttribute("alert_warning", String.format(messages.getString("entity.does.not.exists"), "Trainer", id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+        return "trainer/detail";
+    }
+
+    /**
+     * GET controller for renaming of trainer
+     * @param id of trainer to be renamed
+     * @param model data to be displayed
+     * @return JSP page
+     */
+    @RequestMapping(value = "/rename/{id}", method = RequestMethod.GET)
+    public String rename(@PathVariable long id,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriComponentsBuilder) {
+        log.debug("mvc GET rename({})", id);
+
+        if (trainerFacade.getTrainerWithId(id) == null) {
+            redirectAttributes.addFlashAttribute("alert_danger", String.format(messages.getString("trainer.does.not.exists"), id));
+            return "redirect:" + uriComponentsBuilder.path("/trainer/list").build().encode().toUriString();
+        }
+
+        TrainerRenameDTO trainerRenameDTO = new TrainerRenameDTO();
+        trainerRenameDTO.setTrainerId(id);
+        model.addAttribute("trainerToUpdate", trainerRenameDTO);
+        return "trainer/rename";
+    }
+
+    /**
+     * POST controller for renaming of trainer
+     * @param formBean DTO for rename
+     * @param model data to be displayed
+     * @param id of renamed trainer
+     * @return JSP page
+     */
+    @RequestMapping(value = "/rename/{id}", method = RequestMethod.POST)
+    public String rename(@Valid @ModelAttribute("trainerToUpdate") TrainerRenameDTO formBean,
+                         BindingResult bindingResult,
+                         Model model,
+                         RedirectAttributes redirectAttributes,
+                         UriComponentsBuilder uriComponentsBuilder,
+                         @PathVariable long id) {
+        log.debug("mvc POST rename({})", id);
+        formBean.setTrainerId(id);
```

Comment:
This should come in prefilled, if fields match (between TrainerDTO and TrainerRenameDTO).

Reply:
Tested it and it is not working without setting the id into the formBean.